pika/adapters/select_connection.py

```diff
@@ -290,19 +273,42 @@ def remove_timeout(self, timeout_id):
         :param str timeout_id: The timeout id to remove
 
         """
-        for i in xrange(len(self._timeouts)):
-            t_id, timeout = self._timeouts[i]
-            if t_id == timeout_id:
-                del self._timeouts[i]
-                break
+        try:
+            timeout = self._timeouts.pop(timeout_id)
+            if timeout['deadline'] == self._next_timeout:
+                self._next_timeout = None
+        except KeyError:
+            pass
 
     def start(self):
-        """Start the main poller loop. It will loop here until self.closed"""
-        while self.open:
+        """Start the main poller loop. It will loop here until self.stopping"""
+        LOGGER.debug('Starting IOLoop')
+        self.stopping = False
+        while not self.stopping:
             self.poll()
             self.process_timeouts()
-            self._manage_event_state()
 
+    def stop(self):
+        LOGGER.debug('Stopping IOLoop')
+        self.stopping = True
+        # Sent byte to interupt the poll loop
+        os.write(self.w_interupt, 'X')
+
+    def read_interupt(self, c, e, write_only):
+        # read the interrupt byte
+        os.read(c, 512)
```

Comment:
Is there a possibility that this might get an OSError exception with `EWOULDBLOCK`? Should there be an exception handler that suppresses it just in case?